var/spack/repos/builtin/packages/flecsi/package.py

```diff
@@ -33,50 +55,101 @@ class Flecsi(CMakePackage):
     depends_on('mpi', when='backend=mpi')
     depends_on('mpi', when='backend=legion')
-    depends_on('gasnet@2019.3.0 ~pshm', when='backend=legion')
-    depends_on('legion@ctrl-rep +shared +mpi', when='backend=legion')
-    depends_on('boost@1.59.0: cxxstd=11 +program_options')
+    depends_on('mpi', when='backend=hpx')
+    depends_on('openmpi@3.1.3:3.1.4', when='+openmpi')
+    depends_on('mpich@3.2.1', when='+mpich')
+    depends_on('legion@ctrl-rep +shared +mpi +hdf5', when='backend=legion +hdf5')
+    depends_on('legion@ctrl-rep +shared +mpi', when='backend=legion ~hdf5')
+    depends_on('hpx@1.3.0 cxxstd=14', when='backend=hpx')
+    depends_on('boost@1.70.0: cxxstd=14 +program_options')
     depends_on('metis@5.1.0:')
     depends_on('parmetis@4.0.3:')
+    depends_on('hdf5', when='+hdf5')
     depends_on('caliper', when='+caliper')
     depends_on('graphviz', when='+graphviz')
     depends_on('python@3.0:', when='+tutorial')
     depends_on('llvm', when='+flecstan')
 
+    conflicts('+tutorial', when='backend=hpx')
+#    conflicts('+hdf5', when='backend=hpx')
+#    conflicts('+hdf5', when='backend=mpi')
+
     def cmake_args(self):
-        options = ['-DCMAKE_BUILD_TYPE=debug']
-        options.append('-DCINCH_SOURCE_DIR=' + self.spec['cinch'].prefix)
+        spec = self.spec
+        options = ['-DENABLE_MPI=ON',
+                   '-DENABLE_OPENMP=ON',
+                   '-DCXX_CONFORMANCE_STANDARD=c++17',
+                   '-DENABLE_METIS=ON',
+                   '-DENABLE_PARMETIS=ON',
+                   '-DENABLE_COLORING=ON',
+                   '-DENABLE_DEVEL_TARGETS=ON'
+                  ]
+        options.append('-DCINCH_SOURCE_DIR=' + spec['cinch'].prefix)
+
+        if spec.variants['build_type'].value == 'Debug':
+            options.append('-DCMAKE_BUILD_TYPE=Debug')
+        elif spec.variants['build_type'].value == 'Release':
+            options.append('-DCMAKE_BUILD_TYPE=Release')
 
-        if self.spec.variants['backend'].value == 'legion':
+        if spec.variants['backend'].value == 'legion':
             options.append('-DFLECSI_RUNTIME_MODEL=legion')
-        elif self.spec.variants['backend'].value == 'mpi':
+        elif spec.variants['backend'].value == 'mpi':
             options.append('-DFLECSI_RUNTIME_MODEL=mpi')
+        elif spec.variants['backend'].value == 'hpx':
+            options.append('-DFLECSI_RUNTIME_MODEL=hpx')
         else:
             options.append('-DFLECSI_RUNTIME_MODEL=serial')
-            options.append(
-                '-DENABLE_MPI=OFF',
-            )
+            options.append('-DENABLE_MPI=OFF')
 
+        if '+minimal' in spec:
+            options.append('-DCMAKE_DISABLE_FIND_PACKAGE_METIS=ON')
+        else:
+            options.append('-DCMAKE_DISABLE_FIND_PACKAGE_METIS=OFF')
+        if '+shared' in spec:
+            options.append('-DBUILD_SHARED_LIBSON')
+        else:
+            options.append('-DBUILD_SHARED_LIBSOFF')
+        if '+unittest' in spec:
```

Comment:
Does this option install something different, or simply build the unit tests for testing during installation? If it's the latter, I would drop the variant and make the following change:
```suggestion
        if self.run_tests:
```

Reply:
It doesn't install something different, it's purely to turn on the unit test option at cmake time. How do we tell self.run_tests to be on or off?

Reply:
Does `self.run_tests` just turns on by itself automatically?

Reply:
It is activated depending on whether or not you run `spack install` with the `--test=root` flag. The benefit of this is that it doesn't affect the spec, so if it is already installed, Spack won't reinstall a second copy just because you requested tests.